src/main/java/sirius/search/constraints/RobustQueryParser.java

```diff
@@ -189,12 +189,12 @@ protected QueryBuilder parseOR(LookaheadReader reader) {
      */
     protected QueryBuilder parseAND(LookaheadReader reader) {
         List<QueryBuilder> result = Lists.newArrayList();
-        QueryBuilder subQuery = parseToken(reader);
-        if (subQuery != null) {
-            result.add(subQuery);
-        }
 
         while (!reader.current().isEndOfInput() && !reader.current().is(')')) {
+            QueryBuilder subQuery = parseToken(reader);
```

Comment:
sicher, dass das net die gewichtung (UND vor ODER) zerballert?

aka funktioniert A AND B OR C == (A AND B) OR C sowie A OR B AND C == A OR (B AND C)

Reply:
Mit den beiden Beispielqueries getestet und die Gewischtung scheint zu passen.
Also es wurde jeweils die beiden AND tokens miteinander in einem must constraint verknüpft und dieser mit dem dritten token verodert.